backend/internal/server/apis/schedule/games/manual/games.go

```diff
@@ -0,0 +1,193 @@
+	apis.RegisterHandler(fiber.MethodPut, "/schedule/manual/games/:id", auth.Authenticated, handleUpdateGame)
+	apis.RegisterHandler(fiber.MethodPut, "/schedule/manual/games", auth.Authenticated, handleUpdateGames)
+	apis.RegisterHandler(fiber.MethodDelete, "/schedule/manual/games/:id", auth.Authenticated, handleDeleteGame)
+	apis.RegisterHandler(fiber.MethodDelete, "/schedule/manual/games", auth.Authenticated, handleDeleteGames)
+}
+
+// basic CRUD operations
+// - the caller is allowed to fine tune the schedule by updating the games
+
+func handleCreateGame(c *fiber.Ctx) error {
+	type Dto struct {
+		Game structures.Game `json:"game"`
+	}
+	var dto Dto
+	err := c.BodyParser(&dto)
+	if err != nil {
+		log.Error("Failed to parse request body")
+		return responder.BadRequest(c, fiber.StatusBadRequest, err.Error())
+	}
+	game := dto.Game
+	session := db.GetSession(c)
+	dbGame := mapping.MapGameStructToGameModel([]structures.Game{game})[0]
+	_, err = session.SaveGame(dbGame)
+	if err != nil {
+		log.Error("Failed to save game to the database")
+		return responder.InternalServerError(c, err)
+	}
+	return responder.Ok(c, game)
+}
+
+func handleCreateGames(c *fiber.Ctx) error {
+	type Dto struct {
+		Games []structures.Game `json:"games"`
+	}
+	var dto Dto
+	err := c.BodyParser(&dto)
+	if err != nil {
+		log.Error("Failed to parse request body")
+		return responder.BadRequest(c, fiber.StatusBadRequest, err.Error())
+	}
+	games := dto.Games
+	session := db.GetSession(c)
+	dbGames := mapping.MapGameStructToGameModel(games)
+	_, err = session.SaveGames(dbGames)
+	if err != nil {
+		log.Error("Failed to save games to the database")
+		return responder.InternalServerError(c, err)
+	}
+	return responder.Ok(c, games)
+}
+
+func handleGetGame(c *fiber.Ctx) error {
+	type Dto struct {
+		ID uint `json:"id"`
+	}
+	var dto Dto
+	err := c.BodyParser(&dto)
+	if err != nil {
+		log.Error("Failed to parse request body")
+		return responder.BadRequest(c, fiber.StatusBadRequest, err.Error())
+	}
+	id := dto.ID
+	session := db.GetSession(c)
+	dbGame, err := session.GetGame(id)
+	if err != nil {
+		log.Error("Failed to get game from the database")
+		return responder.InternalServerError(c, err)
+	}
+	game := mapping.MapGameModelToGameStruct([]models.Game{*dbGame})[0]
+	return responder.OkWithData(c, game)
+}
+
+func handleGetGames(c *fiber.Ctx) error {
+	type Dto struct {
+		SeasonID uint `json:"season_id"`
+	}
+	var dto Dto
+	err := c.BodyParser(&dto)
+	if err != nil {
+		log.Error("Failed to parse request body")
+		return responder.BadRequest(c, fiber.StatusBadRequest, err.Error())
+	}
+	seasonID := dto.SeasonID
+	session := db.GetSession(c)
+	dbGames, err := session.GetGames(seasonID)
+	if err != nil {
+		log.Error("Failed to get games from the database")
+		return responder.InternalServerError(c, err)
+	}
+	games := mapping.MapGameModelToGameStruct(*dbGames)
+	return responder.OkWithData(c, games)
```

Comment:
Shouldn't the structures.game be for internal use only and not be the returned data?

Reply:
Talk about this in person or over zoom.

Reply:
I changed it to models.Game, but there was a reason I did it the other way we should talk about.